src/mctpd.c

```diff
@@ -2083,10 +2083,14 @@ static int endpoint_assign_eid(struct ctx *ctx, sd_bus_error *berr,
 	add_peer_route(peer);
 
 	rc = endpoint_send_set_endpoint_id(peer, &new_eid, &req_pool_size);
-	if (rc == -ECONNREFUSED)
+	if (rc == -ECONNREFUSED) {
 		sd_bus_error_setf(
 			berr, SD_BUS_ERROR_FAILED,
 			"Endpoint returned failure to Set Endpoint ID");
+		peer->pool_size = 0;
```

Comment:
This probably warrants a comment. Perhaps:

```c
// we have not yet created the pool route, reset here so that
// remove_peer() does not attempt to remove it
```

Also: why only on `ECONNREFUSED`, and not in the general `rc < 0` case below?